api/src/test/java/org/openmrs/module/ModuleUtilTest.java

```diff
@@ -730,4 +741,40 @@ protected File getEmptyJarDestinationFolder() throws IOException {
 		}
 		return destinationFolder;
 	}
+	
+	@Test
+	public void file2url_shouldReturnNullIfFileIsNull() throws MalformedURLException {
+
+		Assert.assertNull(ModuleUtil.file2url(null));
+	}
+	
+	@Test
+	public void getPackagesFromFile_shouldReturnAnEmptyCollectionIfNonJarFile() {
+		
+		File f = new File(this.getClass()
+			.getResource("/org/openmrs/module/include/test1-1.0-SNAPSHOT.omod").getFile());
+		
+		assertThat(ModuleUtil.getPackagesFromFile(f), is(empty()));
+	}
+	
+	@Test
+	public void getPackagesFromFile_shouldReturnPackagesWithinJar() throws IOException{
```

Comment:
please format the method.

can you please start from top to bottom. first a file that doesnt end in '.jar', then a file that doesnt exist.

the one you wrote is the happy path.

Maybe`ModuleFileParserTest.createJarWithConfigXmlEntry()` is of help for creating an empty jar. Also so we have full control over which path in the code we test we reach and also it makes the test readable.